app/routers/wallet.py

```diff
@@ -16,15 +18,106 @@
 
 admin_url = os.getenv("ACAPY_ADMIN_URL")
 admin_port = os.getenv("ACAPY_ADMIN_PORT")
+# TODO Should the admin_api_key be a dummy variable so the controller doesn't function w/o providing it?
+# This all smells really - this has to be done in a better manner
 admin_api_key = os.getenv("ACAPY_ADMIN_API_KEY")
 is_multitenant = os.getenv("IS_MULTITENANT", False)
 ledger_url = os.getenv("LEDGER_NETWORK_URL")
 
 
-@router.get(
-    "/create-pub-did", tags=["did"], response_model=DidCreationResponse
-)
-async def create_public_did():
+async def __create_did(controller):
+    generate_did_res = await controller.wallet.create_did()
+    if not generate_did_res["result"]:
+        controller.terminate()
+        raise HTTPException(
+            # TODO: Should this return HTTPException, if so which status code?
+            # Check same for occurences below
+            status_code=404,
+            detail=f"Something went wrong.\nCould not generate DID.\n{generate_did_res}",
+        )
+    return generate_did_res
+
+
+async def __post_to_ledger(url, payload):
```

Comment:
We create a ledger service or ledger delegate (not convinced it should be called controller as that's now what it is) and put this method there...

So the wallet context then doesn't know how to call the ledger, it just calls a method post_to_ledger on the ledger service/delegate.